lib_eio_windows/test/test_fs.ml

```diff
@@ -130,69 +125,68 @@ let test_mkdir env () =
   Unix.rmdir "subdir\\nested";
   Unix.rmdir "subdir"
 
-let test_mkdirs env () =
-  let cwd = Eio.Stdenv.cwd env in
-  let nested = cwd / "subdir1" / "subdir2" / "subdir3" in
-  try_mkdirs nested;
-  let one_more = Path.(nested / "subdir4") in
-  (try
-    try_mkdirs one_more
-  with Eio.Io (Eio.Fs.E (Already_exists _), _) -> ());
-  try_mkdirs ~exists_ok:true one_more;
-  try
-    try_mkdirs (cwd / ".." / "outside")
-  with Eio.Io (Eio.Fs.E (Permission_denied _), _) -> ()
-
 let test_symlink env () =
-  (*
+  (* 
     Important note: assuming that neither "another" nor
     "to-subdir" exist, the following program will behave
     differently if you don't have the ~to_dir flag.
 
     With [to_dir] set to [true] we get the desired UNIX behaviour,
     without it [Unix.realpath] will actually show the parent directory
     of "another". Presumably this is because Windows distinguishes
-    between file symlinks and directory symlinks. Fun.
+    between file symlinks and directory symlinks. Fun. 
 
   {[ Unix.symlink ~to_dir:true "another" "to-subdir";
      Unix.mkdir "another" 0o700;
-     print_endline @@ Unix.realpath "to-subdir" |}
+     print_endline @@ Unix.realpath "to-subdir" |} 
   *)
-  let cwd = Eio.Stdenv.cwd env in
-  try_mkdir (cwd / "sandbox");
-  Unix.symlink ~to_dir:true ".." "sandbox\\to-root";
-  Unix.symlink ~to_dir:true "subdir" "sandbox\\to-subdir";
-  Unix.symlink ~to_dir:true "foo" "sandbox\\dangle";
-  try_mkdir (cwd / "tmp");
-  Eio.Path.with_open_dir (cwd / "sandbox") @@ fun sandbox ->
-  try_mkdir (sandbox / "subdir");
-  try_mkdir (sandbox / "to-subdir\\nested");
-  let () =
-    try
+    let symlink_safe ~to_dir src dst = 
+      try
+        Unix.symlink ~to_dir src dst
+      with
+      Unix.Unix_error (Unix.EPERM, _, _) ->
+        Printf.printf "User lacks permission to create symlinks. Skipping symlink creation for %s -> %s.\n" src dst
+        | e -> raise e
+      in
+
+  if Sys.os_type = "Win32" then

```

Comment:
What I had in mind was to simply _skip_ the whole test if symlinking wasn't supported. Something along the lines of 

```ocaml
if not Unix.has_symlink then () else begin
 (* The rest of the test *)
end
```

Reply:
I did something of this nature but thought otherwise. I'd rework on it. Thank you